node/external/blockAPI/metaBlock.go

```diff
@@ -80,6 +84,66 @@ func (mbp *metaAPIBlockProcessor) GetBlockByRound(round uint64, options api.Bloc
 	return mbp.convertMetaBlockBytesToAPIBlock(headerHash, blockBytes, options)
 }
 
+// GetAlteredAccountsForBlock returns the altered accounts for the desired meta block
+func (mbp *metaAPIBlockProcessor) GetAlteredAccountsForBlock(options api.GetAlteredAccountsForBlockOptions) (*common.AlteredAccountsForBlockAPIResponse, error) {
+	headerHash, blockBytes, err := mbp.getHashAndBlockBytesFromStorer(options.GetBlockParameters)
+	if err != nil {
+		return nil, err
+	}
+
+	apiBlock, err := mbp.convertMetaBlockBytesToAPIBlock(headerHash, blockBytes, api.BlockQueryOptions{WithTransactions: true, WithLogs: true})
+	if err != nil {
+		return nil, err
+	}
+
+	txPool := mbp.apiBlockToTxsPool(apiBlock)
+
+	rootHash, err := hex.DecodeString(apiBlock.StateRootHash)
+	if err != nil {
+		return nil, err
+	}
+
+	alteredAccountsOptions := shared.AlteredAccountsOptions{
+		WithCustomAccountsRepository: true,
+		AccountsRepository:           mbp.accountsRepository,
+		// TODO: AccountQueryOptions could be used like options.WithBlockRootHash(..) instead of thinking what to provide
+		AccountQueryOptions: api.AccountQueryOptions{
+			BlockRootHash: rootHash,
+		},
+	}
+
+	alteredAccounts, err := mbp.alteredAccountsProvider.ExtractAlteredAccountsFromPool(txPool, alteredAccountsOptions)
+	if err != nil {
+		return nil, err
+	}
+
+	alteredAccountsAPI := alteredAccountsMapToAPIResponse(alteredAccounts, options.TokensFilter, options.WithMetadata)
+	return alteredAccountsAPI, nil
+}
+
+func (mbp *metaAPIBlockProcessor) getHashAndBlockBytesFromStorer(params api.GetBlockParameters) ([]byte, []byte, error) {
+	if params.RequestType == api.BlockFetchTypeByHash {
+		hashBytes, err := hex.DecodeString(params.Hash)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		headerBytes, err := mbp.getFromStorer(dataRetriever.MetaBlockUnit, hashBytes)
+		return hashBytes, headerBytes, err
```

Comment:
done

Reply:
I think this `if` approach instead of `switch` might be prone to errors.

For example, in case we introduce another `params.RequestType ` this func will always call `mbp.getHashAndBlockBytesFromStorerByNonce(params)`.
If we add a switch with `default` cases some error like: "unknown request type", I think it would be better to catch missing/faulty implementation earlier

Reply:
done. also for shard block (previously the refactoring was only done for meta)